docs/developer/best_practices/commit.rst

```diff
@@ -0,0 +1,372 @@
+existing code. In such case, instead of having single commit with everything
+inside, you can either:
+
+1. first add a new helper: `29f49cbe9dfa <https://github.com/scality/metalk8s/commit/29f49cbe9dfa0b824c818d25d4a2f6965351e65d>`_
+
+2. then use it in new code: `7e47310a8f20 <https://github.com/scality/metalk8s/commit/7e47310a8f20fd49f0ad36707b20e6c2a53df638>`_
+
+Or:
+
+1. first add the new code: `5b2a6d5fa498 <https://github.com/scality/metalk8s/commit/5b2a6d5fa49815180a2effdd37cb58542e83b5a5>`_
+
+2. then refactor the now duplicated code: `ac08d0f53a83 <https://github.com/scality/metalk8s/commit/ac08d0f53a835a0b2bc61c1fe5b7317bf4d6550c>`_
+
+Mixing unrelated changes
+::::::::::::::::::::::::
+
+It is sometimes tempting to do small unrelated changes as you are working on
+something else in the same code area.
+Please refrain to do so, or at least do it in a dedicated commit.
+
+Mixing non-related changes into the same commit makes revert and cherry-pick
+harder (and understanding as well).
+
+The pull request `#1846 <https://github.com/scality/metalk8s/pull/1846>`_ is a good example. It tackles three issues at once: `#1830 <https://github.com/scality/metalk8s/issues/1830>`_
+and `#1831 <https://github.com/scality/metalk8s/issues/1831>`_ (because they are similar) and `#839 <https://github.com/scality/metalk8s/issues/839>`_ (because it was making the other
+changes easier), but it uses distincts commits for each issue.
+
+How to write a commit message
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Why do we need commit messages
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+After comments in the code, commit messages are the easiest way to find context
+for every single line of code: running ``git blame`` on a file will give you,
+for each line, the identifier of the last commit that changed the line.
+
+Unlike a comment in the code (which applies to a single line or file), a commit
+message applies to a logical change and thus can provide information on the
+design of the code and why the change was done. This makes commit messages a
+part of the code documentation and makes them helpful for other developers to
+understand your code.
+
+Last but not least: commit messages can also be used for automating tasks such
+as issue management.
+
+Note that it is important to have all the necessary information in the commit
+message, instead of having them (only) in the related issue, because:
+
+- the issue can contain troubleshooting/design discussion/investigation with a
+  lot of back and forth, which makes hard to get the gist of it.
+
+- you need access to an external service to get the whole context, which goes
+  against one of biggest advantage of the distributed SCM (having all the
+  information you need offline, from your local copy of the repository).
+
+- migration from one tracking system to another will invalidate the
+  references/links to the issues.
+
+Anatomy of a good commit message
```

Comment:
Can we provide a template? Users could "install" it locally, with `git config commit.template` :)

Reply:
I don't have one ready right now.

Also, I'm worried it would be more of a hindrance than helpful since you'll always have text to delete (because you can hardly generate a useful default).

Reply:
We can use comments for non-required content, I guess.

Something like:

```
topic: subject (stay below 50 chars)

# Commit message body
# Explains:
#   - why the change,
#   - what was changed,
#   - discarded alternatives

# Commit message footer
# References:
#   - issues fixed by this change, using e.g. `Fixes: #XXXXX`
#   - issues linked to this change, using e.g. `Refs: #XXXXX`
#   - external links, using e.g. `See: kubernetes/kubernetes/issues/XXXXX`
```

Reply:
Where would that template lives?
- Inline in this doc?
- Or somewhere in the repo?

(still not really convinced of the help of this template though ^^)